### PR TITLE
Feat/parameterized macros

### PR DESCRIPTION
**Pull request by piernov**
Supersedes #56

### CODE REVIEW

File: src/brainfuck/MacroParser.java [hunk 1]
```diff
@@ -109,6 +110,38 @@ public MacroParser(Stream<String> stream) {
...
+	 * @param params	parameters for macro expansion: params[0] is the macro name, params[1] is the repetition count
+	 * @return macro recursively expanded (optionally multiple times)
+	 */
+	private Stream<String> writeMacroBody(String[] params) {
+		String name = params[0];
+		int repeat = 1;
+		if (params.length > 1) repeat = Integer.parseInt(params[1]);
+
+		Macro macro = macros.get(name);
+		return IntStream.range(0, repeat).mapToObj(i -> macro.getBody().flatMap(this::parseLine)).flatMap(t->t);
```
Comment (sticmac) [inline src/brainfuck/MacroParser.java]: Could you put comments about this?
And write it on several lines?

**Comment by sticmac** [comment]
Maybe `MACRO_ARGS` is not needed anymore?

**Review by sticmac — CHANGES_REQUESTED** [review]
Some modifications to do still, in my opinion.
(See my comments).

**Review by sticmac — APPROVED** [review]

**Comment by piernov** [comment]
@Tordesillas @sticmac @NassimBounouas 
Do we want to use this implementation in the end?
(if so, #58 must be reworked)

**Comment by sticmac** [comment]
@piernov I can't think of any better solution, so i think this one could be great.

**Review by NassimBounouas — APPROVED** [review]

**Pull request merged**